app/src/main/java/com/example/movies/MovieListRecyclerAdapter.kt

```diff
@@ -0,0 +1,67 @@
+package com.example.movies
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import me.zhanghai.android.materialratingbar.MaterialRatingBar
+
+class MovieListRecyclerAdapter(
+    private val listener: () -> Unit,
+    private val movies: List<MovieData>
+) :
+    RecyclerView.Adapter<MovieListRecyclerAdapter.MovieListViewHolder>() {
+    class MovieListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        var movieName: TextView? = null
+        var moviePic: ImageView? = null
+        var movieAge: TextView? = null
+        var movieLike: ImageView? = null
+        var ratingBar: MaterialRatingBar? = null
+        var movieReviews: TextView? = null
+        var movieGenre: TextView? = null
+        var duration: TextView? = null
+
+        init {
+            movieName = itemView.findViewById(R.id.movie_name)
+            moviePic = itemView.findViewById(R.id.movie_pic)
+            movieAge = itemView.findViewById(R.id.movie_age)
+            movieLike = itemView.findViewById(R.id.movie_like)
+            ratingBar = itemView.findViewById(R.id.ratingBar)
+            movieReviews = itemView.findViewById(R.id.movie_reviews)
+            movieGenre = itemView.findViewById(R.id.movie_genre)
+            duration = itemView.findViewById(R.id.duration)
+        }
```

Comment:
lets refactor it from nullable type to `val` and non null as in the previous adapter